### PR TITLE
Updates for DG, Logic Class diagram, Model Class diagram

### PR DESCRIPTION
**Pull request by minpyaemoe**
- Updated DG
- Update Logic Class Diagram
- Update Model Class Diagram

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -504,56 +504,151 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
...
 .Activity diagram for removing a reminder
-image::DeleteReminderActivityDiagram.png[width=75%]
+image::DeleteReminderActivityDiagram.png[width=90%]
 
 [NOTE]
-The negative index from user input will leads to invalid command format error.
+The negative index from user input will leads to invalid command format error as shown in the above diagram.
 
-The below sequence diagram summarize the interactions between different components when user enter `DeleteReminderCommand`.
+Moreover, the below sequence diagram summarizes the interactions between different components when user enter `DeleteReminderCommand`.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: the **following**

when **the** user **enters**

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -504,56 +504,151 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
...
+image::DeleteReminderSequenceDiagram.png[width=100%]
+
+The following are the brief explanation of <<DeleteReminderSequenceDiagram>>
+
+1. The user enters `reminder delete 1`
+2. `SpendingBookParser#parseCommand()` called by `LogicManager`.
+3. This creates `ReminderCommandParser` and it checks if the given user input is to add a reminder, delete a reminder or invalid.
+4. Next, `DeleteReminderCommandParser` is created and it validates the parameters provided by `ReminderCommandParser`.
+5. After successful validation, `DeleteReminderCommand` is created and passed back to callers until it reaches to `Logic Manager`.
+6. `LogicManger` then executes the command, removing the specified reminder from the model through `Model#deleteReminder()`.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: 3. This creates `ReminderCommandParser` and it checks if the given user input is to add or delete a reminder or is invalid.

**LogicManager**

**Review by choongyx — APPROVED** [review]

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -504,56 +504,151 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
...
 
 [NOTE]
-The above diagram also highlights how logic and model components interact with each other while deleting a reminder.
+The above sequence diagram only shows how a `DeleteReminderCommand` is created and executed. It does not include how a reminder is removed from the storage component and how the UI component is updated.
+
+==== Limitations
+
+1. The date value of the `deadline` attribute is limited to before the end of next year and must be in the future. This prevent users from entering reminder of overdue tasks as inputs.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: **prevents** users from entering overdue tasks as a reminder.

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -504,56 +504,151 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
...
+
+|Store reminders in a json file separated from other information. +
+ +
+*Pros:* +
+Improve performance while importing and exporting reminders. +
+Lesser data loss in case of file corruption. +
+ +
+*Cons:* +
+Could complicate future implementation of importing and exporting reminders.
+`import` and `export` commands now need to manges two file at the same time.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: **manages**

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -504,56 +504,151 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
...
+
+|===
+
+Alternative 1 from above table, is currently implemented in MoneyGoWhere (v1.4) since current version of `Import` and `Export` Commands has not be integrated with reminders yet.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: This sentence is very out of place

**Review by Nanosync — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -504,56 +504,151 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
...
+image::AddReminderSequenceDiagram.png[pdfwidth="95%",width="100%"]
 
 [NOTE]
-The above sequence diagram demonstrates how a new reminder is constructed from valid user input.
+The above sequence diagram only demonstrates how an `AddReminderCommand` is constructed from valid user input and how it is executed. It does not show how a reminder is added into the `Storage` component and how the `UI` components are updated.
+
+The following are the brief explanation of <<AddReminderSequenceDiagram>>
+
+1. The user enters `reminder add d/today m/Pay bill`.
+2. `SpendingBookParser#parseCommand()` is called by `LogicManger`
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: Here also

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -504,56 +504,151 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
...
+3. This creates `ReminderCommandParser` and it checks if the given user input is to add a reminder, to delete a reminder or invalid.
+4. Next, `AddReminderCommandParser` is created and it validates the parameters provided by `ReminderCommandParser`.
+5. After successful validation, `AddReminderCommand` is created and passed back to callers until it reaches to `LogicManager`.
+6. `LogicManger` then executes the command, adding the new reminder to the model through `Model#addReminder()`.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: Here too

**Review by Nanosync — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (minpyaemoe) [reply docs/DeveloperGuide.adoc]: amended

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (minpyaemoe) [reply docs/DeveloperGuide.adoc]: amended

File: docs/DeveloperGuide.adoc (hunk 4 above)
Comment (minpyaemoe) [reply docs/DeveloperGuide.adoc]: amended

File: docs/DeveloperGuide.adoc (hunk 6 above)
Comment (minpyaemoe) [reply docs/DeveloperGuide.adoc]: managed

File: docs/DeveloperGuide.adoc (hunk 7 above)
Comment (minpyaemoe) [reply docs/DeveloperGuide.adoc]: amended

File: docs/DeveloperGuide.adoc [hunk 8]
```diff
@@ -504,56 +504,149 @@ In contrast, for Alternative 2, manipulating the internal elements of the list d
 // tag::reminder[]
 === Reminder feature
 
-With this reminder feature, users can set the reminders of their tasks with deadlines and delete them when completed.
+This feature allows users to set reminders for their tasks with certain deadline and delete them when respective tasks are completed.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: with a specified deadline

**Review by Nanosync — APPROVED** [review]

**Pull request merged**